config/initializers/rack_attack.rb

```diff
@@ -20,19 +40,23 @@ def track_and_return_ip
     end
 
     throttle("api_throttle", limit: 3, period: 1) do |request|
-      if request.path.starts_with?("/api/") && request.get?
+      api_endpoint = request.path.starts_with?("/api/")
+      if api_endpoint && request.get? && !admin_api_key?(request)
         request.track_and_return_ip
       end
     end
 
     throttle("api_write_throttle", limit: 1, period: 1) do |request|
-      if request.path.starts_with?("/api/") && (request.put? || request.post? || request.delete?)
+      api_endpoint = request.path.starts_with?("/api/")
+      if api_endpoint && (request.put? || request.post? || request.delete?)
         Honeycomb.add_field("user_api_key", request.env["HTTP_API_KEY"])
-        ip_address = request.track_and_return_ip
-        if request.env["HTTP_API_KEY"].present?
-          "#{ip_address}-#{request.env['HTTP_API_KEY']}"
-        elsif ip_address.present?
-          ip_address
+        unless admin_api_key?(request)
+          ip_address = request.track_and_return_ip
+          if request.env["HTTP_API_KEY"].present?
+            "#{ip_address}-#{request.env['HTTP_API_KEY']}"
+          elsif ip_address.present?
+            ip_address
+          end
```

Comment:
I noticed I can't use a guard clause along the lines of `return unless admin_api_key?` here, so I added this `unless` check. It starts to feel a little too nested in here, so I'm open to any reorganization suggestions in this area if anyone feels like it should be structured differently 🙂

Reply:
I agree that the nesting might not be ideal, but I also can't think of a cleaner alternative. This might be our best option for what we have now.